File: Java/PostExample/app/src/main/java/com/tmdhoon/postexample/MainActivity.java
package com.tmdhoon.postexample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import com.tmdhoon.postexample.databinding.ActivityMainBinding;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private Editable postId;

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        postId = binding.EditText.getText();

        PostAPI postAPI = ApiProvider.getRetrofit().create(PostAPI.class);

        binding.btClick.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                postAPI.getPost(postId).enqueue(new Callback<PostResponse>() {
                    @Override
                    public void onResponse(Call<PostResponse> call, Response<PostResponse> response) {
                        if(response.isSuccessful()){
                            binding.textView.setText(response.body().getContent());
                        }else{
                            Toast.makeText(MainActivity.this, "Error!", Toast.LENGTH_SHORT).show();
                        }
                    }

                    @Override
                    public void onFailure(Call<PostResponse> call, Throwable t) {
                        Toast.makeText(MainActivity.this, "통신 실패", Toast.LENGTH_SHORT).show();
                    }
                });
            }
        });

    }
}

File: Kotlin/Apipractice/app/src/main/java/com/tmdhoon/apipractice/MainActivity.kt
package com.tmdhoon.apipractice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.bumptech.glide.Glide
import com.tmdhoon.apipractice.databinding.ActivityMainBinding
import com.tmdhoon.apipractice.response.UserResponse
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        RetrofitBuilder.githubApi.getGithubInfo().enqueue(object : Callback<UserResponse> {

            override fun onResponse(
                call: Call<UserResponse>,
                response: Response<UserResponse>
            ) {
                if(response.isSuccessful) {
                    val userInfo = response.body()

                    binding.tvFollowerCount.text = userInfo!!.followers.toString()
                    binding.tvFollowingCount.text = userInfo!!.following.toString()
                    binding.tvName.text = userInfo!!.login
                    binding.tvDescription.text = userInfo!!.bio

                    getImageURL(userInfo.avatar_url)
                }

            }

            fun getImageURL(imageURL: String){
                Glide.with(applicationContext)
                    .load(imageURL)
                    .into(binding.ivProfileImage)
            }

            override fun onFailure(call: Call<UserResponse>, t: Throwable) {

            }

        })


    }
}